math/vector_cross_product.cpp

```diff
@@ -0,0 +1,111 @@
+ * first vector and the third row with the direction ratios of the second vector.
+ * The magnitude of a vector is it's value expressed as a number.
+ * Let the direction ratios of the first vector, P be: a, b, c
+ * Let the direction ratios of the second vector, Q be: x, y, z
+ * Therefore the calculation for the cross product can be arranged as:
+ * P x Q:
+ *  	1	1	1
+ *  	a	b	c
+ *  	x	y	x
+ *
+ * @algorithm The direction ratios (DR) are calculated as follows:
+ *  	1st DR, J:  (b * x) - (c * y)
+ *  	2nd DR, A: -((a * x) - (c * x))
+ *  	3rd DR, N:  (a * y) - (b * x)
+ *
+ * Therefore, the direction ratios of the cross product are: J, A, N
+ * The following C++ Program calculates the direction ratios of the cross products of two vector.
+ * The program uses a function, cross() for doing so.
+ * The direction ratios for the first and the second vector has to be passed one by one seperated by a space character.
+ *
+ * Magnitude of a vector is the square root of the sum of the squares of the direction ratios.
+ *
+ * @example An example of a running instance of the executable program:
+ *
+ * 	Pass the first Vector: 1 2 3
+ *	Pass the second Vector: 4 5 6
+ *	The cross product is: -3 6 -3
+ *	Magnitude: 7.34847
+ *
+ * @author [Shreyas Sable](https://github.com/Shreyas-OwO)
+ */
+
+#include <iostream>
+#include <array>
+#include <cmath>
+
+namespace math {
+	/**
+	 * @namespace math
+	 * @brief Math algorithms
+	 */
+	namespace vector_cross {
+		/**
+		 * @namespace vector_cross
+		 * @brief Functions for Vector Cross Product algorithms
+		 */
+
+		/**
+		 * @brief Function to calculate the cross product of the passed arrays containing the direction ratios of the two mathematical vectors.
+		 * @param A type: std::array<double, 3> description: contains the direction ratios of the first mathematical vector.
+		 * @param B type: std::array<double, 3> description: contains the direction ration of the second mathematical vector.
+		 * @returns type: std::array<double, 3> description: contains the direction ratios of the cross product.
+		 */
+		std::array<double, 3> cross(std::array<double, 3> A, std::array<double, 3> B) {
+			std::array<double, 3> product;
+			/// Performs the cross product as shown in @algorithm.
+			product[0] = (A[1] * B[2]) - (A[2] * B[1]);
+			product[1] = -((A[0] * B[2]) - (A[2] * B[0]));
+			product[2] = (A[0] * B[1]) - (A[1] * B[0]);
+			return product;
+		}
+
+		/**
+		 * @brief Calculates the magnitude of the mathematical vector from it's direction ratios.
+		 * @param vec type: std::array<double, 3> description: an array containing the direction ratios of a mathematical vector.
+		 * @returns type: double description: the magnitude of the mathematical vector from the given direction ratios.
+		 */
+		double mag(std::array<double, 3> vec) {
+			double magnitude = sqrt((vec[0] * vec[0]) + (vec[1] * vec[1]) + (vec[2] * vec[2]));
+			return magnitude;
+		}
+	} /// namespace vector_cross
+} /// namespace math
+
```

Comment:
please add a test function as shown in other codes and use `assert` function to validate the code for sample inputs.